lib/Serialization/Deserialization.cpp

```diff
@@ -2586,21 +2586,17 @@ ModuleFile::getDeclCheckedImpl(DeclID DID) {
           vjpDecl = cast<FuncDecl>(getDecl(vjpDeclId));
         }
 
-        SmallVector<AutoDiffParameter, 4> parameters;
-        SourceLoc loc;
-        for (auto paramValue : paramValues) {
-          auto parameter = paramValue & 0x01
-            ? AutoDiffParameter::getSelfParameter(loc)
-            : AutoDiffParameter::getIndexParameter(loc, paramValue >> 1);
-          parameters.push_back(parameter);
-        }
-        // TODO: Deserialize CheckedParameterIndices.
+        llvm::SmallBitVector parametersBitVector(parameters.size());
+        for (unsigned i = 0; i < parameters.size(); i++)
```

Comment:
```suggestion
        for (unsigned i : indices(parameters))
```

Reply:
Done in https://github.com/apple/swift/pull/21837/commits/6fe2c6458258a66d31c72730d89aed36f297996d. Also done in `SerializeSIL.cpp` and `DeserializeSIL.cpp`.